File: src/com/kedzie/vbox/api/ISessionStateChangedEvent.java
package com.kedzie.vbox.api;

import com.kedzie.vbox.api.jaxb.SessionState;
import com.kedzie.vbox.soap.KSOAP;

public interface ISessionStateChangedEvent extends IEvent {
	@KSOAP(cacheable=true) public SessionState getState();
}

File: vbox-app/src/main/java/com/kedzie/vbox/api/ISessionStateChangedEvent.kt
package com.kedzie.vbox.api

import com.kedzie.vbox.api.jaxb.SessionState
import com.kedzie.vbox.soap.Cacheable
import com.kedzie.vbox.soap.Ksoap
import com.kedzie.vbox.soap.KsoapProxy

@KsoapProxy
@Ksoap
interface ISessionStateChangedEvent : IMachineEvent {
    @Cacheable("State")
	suspend fun getState(): SessionState
}